### DISCUSSION TITLE
SmartHome Funkstick für HomeMatic IP = HmIP-RFUSB oder HmIP-RFUSB-TK?

### DISCUSSION THREAD

**Discussion by carusoe** [OpenCCU/OpenCCU#1651] [🇩🇪 Fragen&Antworten] [answered]
Hallo,

ich habe gerade unter der URL https://www.smarthome.de/geraete/smarthome-funkstick-fuer-homematic-ip-weiss den Stick gesehen und mich gefragt, ob dieser Stick baugleich zum HMIP-Stick (https://de.elv.com/elv-homematic-ip-arr-bausatz-rf-usb-stick-fuer-alternative-steuerungsplattformen-hmip-rfusb-fuer-smart-home-hausautomation-152306?fs=2353872394&c=691) ist. Ich möchte den einfach mal ausprobieren, ob ich meine Homatic-Geräte und Sensoren auch damit steuern kann. Ich setze gerade eine HA-Installation mit dem RaspberryMatic-Add-On auf einem Raspberry PI 3b+ auf und suche noch nach dem idealen Setup für eine Wohnung, die sehr lang gestreckt ist. Empfehlungen sind sehr willkommen. :-) 

Zusatzfrage: Hat dieser ELV-Stick/Telekom Stick eine höhere Funkreichweite als der HM-MOD-RPI-PCB (https://de.elv.com/elv-homematic-komplettbausatz-funkmodul-fuer-raspberry-pi-hm-mod-rpi-pcb-fuer-smart-home-hausautomation-142141?fs=3697633989)?

Danke schon einmal für Eure Einschätzungen und Antworten. 

Carusoe

**Comment by LC7894** [comment]
Hallo @carusoe, 

ich habe mich vor Kurzem genau das Gleiche gefragt wie du und mich letztendlich für den "Telekom-Stick" entschieden. Dieser ist exakt der Gleiche wie der Stick von ELV mit dem Unterschied, dass du ihn nicht selbst zusammenbauen/-löten musst. Allerdings habe ich irgendwo gelesen, dass es bezüglich Updates ggf. Probleme geben könnte. Zu diesem Punkt kann ich dir nichts genaueres sagen.

In meiner HA-Installation funktioniert der Stick mit dem RaspberryMatic-Add-On einwandfrei. Zur Funkreichweite kann ich dir ebenfalls nichts genaueres sagen.

**Reply by carusoe** [reply]
Hallo @LC7894,

wie bist Du denn vorgegangen bzw. welche Anleitungen hast Du benutzt? Mein Problem ist, dass ich mich erst seit ca. zwei Wochen mit dem Thema auseinandersetze und nicht noch nicht die Basics "drauf habe". Daher wäre toll, wenn Du oder die anderen mir eine Anleitung, die Schritt-für-Schritt erklärt, was in welcher Reihenfolge zu tun ist. Die Anleitung von Jens zur Installation des And-Ons habe ich bereits, aber verstehen tu ich das letzte Drittel nicht wirklich... :-S ;-)

**Reply by LC7894** [reply]
Was meinst du denn mit dem letzten Drittel? Also ab welchem Schritt genau kommst du nicht weiter? :)

Bezüglich der Programmierung der Ausfallautomatisierung (https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#add-a-home-assistant-automation-to-connect-to-the-raspberrymatic-addon), Schritte 1-2, kannst du auch dieser Anleitung folgen: 
https://flashlog.de/2018/12/home-assistant-verbindung-zur-raspberrymatic-homematic-ccu-pruefen/. Das hat mir weitergeholfen, da ich mich mit der CCU anfangs gar nicht ausgekannt habe.

**Reply by carusoe** [reply]
Also ich versuche wenigstens die Anleitungen zu verstehen, indem ich sie mir zuerst durchlese... ;-) Daher habe ich noch nicht mit der Installation angefangen... Ich schwanke noch, ob ich den Raspi 3b+ oder den Raspi 4 einsetze (beide habe ich zuhause rumfliegen). 

Die Schritte ab "Home Assistant Integration Setup" (https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant#home-assistant-integration-setup) sind mir nicht klar, was da gemacht wird... Ich glaube ich muss es einfach mal ausprobieren... XD Aber vielleicht kann es mir jemand erklären? :-)

**Reply by LC7894** [reply]
Das macht Sinn ;-) Wenn du Home Assistant nicht nur für RaspberryMatic benutzen möchtest, würde ich dir in jedem Fall einen Raspi 4 empfehlen. Ich habe inzwischen so viele Add-Ons und Integrationen auf Home Assistant laufen, dass ich sehr froh über den Raspi 4 bin.

Zu den Schritten ab dem Integration Setup kannst du dir mal das Video anschauen: https://youtu.be/G85kOZtt_Uk?t=358 (ab dem Timestamp bis 7:05). Dann siehst du, wo du die Config-Einträge machen musst und wo du die Ports freigeben musst. Allerdings geht es in dem Video nicht um das RaspberryMatic Add-On, sondern um das alte Homematic Add-On. Deswegen schau dir das Video nur zur Orientierung an und folge beim Rest der Anleitung. So musst du z.B. beachten, dass du als Host nicht `core-homematic` eingeben darfst, sondern den Namen des Add-Ons. Wie du das findest, ist in der von dir verlinkten Anleitung erklärt. Reicht dir das soweit? 🙂

**Reply by carusoe** [reply]
Hallo @LC7894,

vielen Dank für die Erklärung. Das reicht mir erst einmal. :-D

**Comment by bronkoo** [comment]
> ich habe gerade unter der URL https://www.smarthome.de/geraete/smarthome-funkstick-fuer-homematic-ip-weiss den Stick gesehen und mich gefragt, ob dieser Stick baugleich zum HMIP-Stick (https://de.elv.com/elv-homematic-ip-arr-bausatz-rf-usb-stick-fuer-alternative-steuerungsplattformen-hmip-rfusb-fuer-smart-home-hausautomation-152306?fs=2353872394&c=691) ist. 

Hallo @carusoe ,
dazu gab es im letzten [Release eine schöne Aufklärung](https://github.com/jens-maus/RaspberryMatic/releases/tag/3.61.7.20211218), auch für mich:

- integrated a new 4.4.16 coprocessor firmware for the USB-based HmIP-RFUSB rf module sold by eQ3/ELV. This new "dualcopro" firmware released by eQ3 finally supports to use the **HmIP-RFUSB** as a dual BidCos/HomeMatic and homematicIP RF communication device and bumps it to the same feature set like the GPIO-based RPI-RF-MOD. This means, that the HmIP-RFUSB USB stick can now finally communicate with traditional "HomeMatic" devices as well as "homematicIP" devices and also supports all "Advanced homematicIP features" which are required to connect a HmIP-HAP or HmIPW-DRAP LAN gateway device.
 **PLEASE NOTE**: This new dualcopro firmware is only available for the HmIP-RFUSB sticks distributed by eQ3/ELV and NOT for the **HmIP-RFUSB-TK** which is the so-called "Telekom" version of the same RFUSB stick but meant to be used/supported only in combination with a Telekom SpeedPort router.


Für reines **Homatic IP** funktionieren also beide wie gehabt, **Homatic** nur ab Release 3.61.7.20211218 und nur mit der **eQ3/ELV** Version...

**Reply by carusoe** [reply]
Hallo,
das ist ja interessant. Wie kann ich die Firmware-Version herausfinden?

**Reply by jens-maus** [reply]
Geh unter `Einstellungen -> Geräte` auf das `HmIP-RFUSB` Gerät und da sollte dann die Firmwareversion bei den Geräteeinstellungen mit ausgegeben werden. Zusätzlich kannst du auf der Zentrale einen Blick in die `/var/hm_mode` Datei werfen, da sollten alle relevanten Informationen zum Funkmodul (inkl. der Firmwarestände) aufgelistet werden.

**Reply by jens-maus** [reply]
> Für reines **Homematic IP** funktionieren also beide wie gehabt, **Homatic** nur ab Release 3.61.7.20211218 und nur mit der **eQ3/ELV** Version...

Auch das stimmt nur bedingt, denn wenn man einen `HmIP-HAP` zur Reichweitenverlängerung oder homematicIP-Wired via anbinden eines `HmIPW-DRAP` nutzen möchte kann man auch nicht auf den `HmIP-RF-USB-TK` der Telekom setzen. Und auch die ganzen neueren "homematicIP Advanced Routing" Features bringt die alte Firmware des `HmIP-RF-USB-TK` auch nicht mit. Ergo, spart man da IMHO an der falschen Stelle wenn man nur des Geldes wegen auf den `HmIP-RF-USB-TK` setzt statt "das Original" `HmIP-RFUSB` von eQ3/ELV direkt zu kaufen.

**Reply by bronkoo** [reply]
_Danke für die Korrektur und Aufklärung._

**Reply by totti4ever** [reply]
> > Für reines **Homematic IP** funktionieren also beide wie gehabt, **Homatic** nur ab Release 3.61.7.20211218 und nur mit der **eQ3/ELV** Version...
> 
> Auch das stimmt nur bedingt, denn wenn man einen `HmIP-HAP` zur Reichweitenverlängerung oder homematicIP-Wired via anbinden eines `HmIPW-DRAP` nutzen möchte kann man auch nicht auf den `HmIP-RF-USB-TK` der Telekom setzen. Und auch die ganzen neueren "homematicIP Advanced Routing" Features bringt die alte Firmware des `HmIP-RF-USB-TK` auch nicht mit. Ergo, spart man da IMHO an der falschen Stelle wenn man nur des Geldes wegen auf den `HmIP-RF-USB-TK` setzt statt "das Original" `HmIP-RFUSB` von eQ3/ELV direkt zu kaufen.

@jens-maus, was ist denn der günstigste Weg, etwas zu kaufen, was nicht mehr gelötet werden muss und was ich an meinen Linux-Server einfach anschließen kann, um es mit der Docker-Version von Raspberrymatic zu verwenden?

Ich finde fertig zusammengelötete Sticks für 50€, den Löt-Bausatz für 35€...gibts noch günstigeres oder ist das der empfohlene Weg außerhalb des rPi Universums?

**Comment by MichaelN0815** [comment]
Der Telekom Stick wird wohl keine Firmware Updates bekommen.

**Reply by carusoe** [reply]
Wenn die baugleich sind, wieso kann der Telekom-Stick keine Updates bekommen. Ich werde ihn ja nicht in einem T-Com-Router einsetzen, sondern in meinem Raspberry Pi..

**Reply by jens-maus** [reply]
Weil die `HmIP-RFUSB-TK` Sticks ausschließlich für die Telekom bzw. für deren SpeedPort Router gefertigt und auch nur exlusiv von der Telekom verkauft werden und diese so entsprechend gebrandet sind und die Telekom es nicht will das unterschiedliche Firmware-Stände hier+da rumfliegen im Internet. Ist also eine reine administrative bzw. rechtliche Angelegenheit warum für die `HmIP-RFUSB-TK` Sticks es keine Firmwarefiles mit RaspberryMatic ausgeliefert werden bzw. es die auch nirgends gibt.

**Comment by jens-maus** [comment, marked as answer]
Ich versuche nun mal von meiner Seite aus das Thema bzgl. `HmIP-RFUSB` vs. `HmIP-RFUSB-TK` final zu beantworten.

> ich habe gerade unter der URL https://www.smarthome.de/geraete/smarthome-funkstick-fuer-homematic-ip-weiss den Stick gesehen und mich gefragt, ob dieser Stick baugleich zum HMIP-Stick (https://de.elv.com/elv-homematic-ip-arr-bausatz-rf-usb-stick-fuer-alternative-steuerungsplattformen-hmip-rfusb-fuer-smart-home-hausautomation-152306?fs=2353872394&c=691) ist. [...] Empfehlungen sind sehr willkommen. :-)

Also ich kann von technischer Sicht nur dazu raten direkt zum `HmIP-RFUSB` von eQ3/ELV zu greifen und die Finger von der Telekomvariante `HmIP-RFUSB-TK` zu lassen, denn bei dieser Variante handelt es sich um eine spezielle Hardwareversion des USB-Funksticks der nur für die SpeedPort-Router der Telekom verkauft und entsprechend mit support&co versorgt werden. Deshalb gibt es auch dafür keine neuere 4.4.x Firmware die mit allen neuesten/aktuellsten homematicIP Features daherkommt wie die Firmware die es seit der aktuellen RaspberryMatic Version für den `HmIP-RFUSB` ausgeliefert wird. Bei diesen neueren "homematicIP Features" handelt es sich um das sognannte "homematicIP Advanced Routing" das mit dieser 4.4.x Firmware substantiell verbessert bzw. eingeführt wurde. Und auch das Anbinden eines `HmIP-HAP` als Reichweitenverlängerungs-Gateway bzw. das Anbinden eines `HmIPW-DRAP`  als homematicIP-Wired Gateway ist nur mit dieser Firmware-Version möglich. D.h. mit einem `HmIP-RFUSB-TK` hat man diese Funktionalitäten nicht und bei weiteren kommenden Firmware-Updates die ggf. auch weitere Verbesserungen bzw. Bugfixes mit sich bringen bleibt man diesbzgl. dann aussen vor. Ich bleibe also dabei, aus heutiger technischer Sicht sollte man entweder zu einem `RPI-RF-MOD` oder einem `HmIP-RFUSB` greifen wenn man die größstmöglichste Kompatibilität und Funktionalität haben will bzw. sichergestellt haben will und auch aus administrativer Sicht sollte man die Finger vom `HmIP-RFUSB-TK` lassen und nicht diesen nur kaufen weil der vermeintlich ein paar Euro günstiger ist als der Bausatz von eQ3/ELV. Und wer ggf. den Bausatz von eQ3/ELV wegen notwendiger Lötarbeiten meidet, dem sei gesagt, das hier nur eine handvoll Lötpunkte durchgeführt werden müssen und dafür im Bekanntenkreis jemanden zu finden der einem das mal schnell zusammenlötet sollte nicht gar so schwierig sein.
 
> Zusatzfrage: Hat dieser ELV-Stick/Telekom Stick eine höhere Funkreichweite als der HM-MOD-RPI-PCB (https://de.elv.com/elv-homematic-komplettbausatz-funkmodul-fuer-raspberry-pi-hm-mod-rpi-pcb-fuer-smart-home-hausautomation-142141?fs=3697633989)?

Die Funkreichweite sollte bei beiden entsprechend gleich sein (zumindest was die Specs angeht). Wie sich das in der Praxis dann verhält muss man natürlich im Einzelfall sehen. Bei beiden sollte man ohnehin mitunter eine externe Antenne verlöten wenn man die maximale Reichweite damit erreichen möchte. Und abschließend sollte man ohnehin nicht mehr zum `HM-MOD-RPI-PCB` greifen, weil dieses nicht nur wie der `HmIP-RFUSB-TK` von der Telekom mit einer veralteten Firmware daherkommt, sondern auch vom Funkmodultype auf alte Hardware setzt für die es definitiv nie mehr ein FirmwareUpdate geben wird.

> 
> Danke schon einmal für Eure Einschätzungen und Antworten.
> 
> Carusoe

**Reply by bronkoo** [reply]
Hier geht es ja um die **CCU**. Aus meiner bescheidenen Sicht eines **Homatic** Anfängers, erscheint der `HM-MOD-RPI-PCB`auf einem `Raspberry Pi Zero (W)` als eine günstige und vernünftige Lösung für ein [Lan-Gateway](https://github.com/jens-maus/RaspberryMatic/wiki/Experten-Features#lan-gateway-betrieb). Sammele aber gerade noch Langzeiterfahrung in Verbindung mir der Nutzung über ein OpenVPN Tunnel an einem zweiten Standort ...

**Reply by carusoe** [reply]
Hallo @jens-maus,

das heisst meinen HM-MOD-RPI-PCB sollte ich dann auf lange Sicht durch einen  RPI-RF-MOD oder einem HmIP-RFUSB ersetzen? Was empfiehlst Du, besser vor der Installation oder kann ich mein dann bestehendes System auf den RPI-RF-MOD "migrieren"?

**Reply by jens-maus** [reply]
Das kannst du prinzipiell jeder zeit (um)migrieren, ja. Allerdings, wenn du jetzt ohnehin gerade am Anfang bist würde ich dir gleich zu einem `RPI-RF-MOD` oder `HmIP-RFUSB` mit entsprechender externen Antenne dran raten. Das `HM-MOD-RPI-PCB` hat wirklich nur noch einen sehr begrenzten Einsatzzweck diesbzgl.

**Reply by carusoe** [reply]
Hallo,
das heisst, wenn ich die Artikel in dieser Diskussion interpretiere, dann kann ich den HmIP-MOD auch mit dem Raspi 4 einbauen?

**Reply by AlexanderDedy** [reply]
Hallo du hast ‚‚Bei beiden sollte man ohnehin mitunter eine externe Antenne verlöten wenn man die maximale Reichweite damit erreichen möchte‘‘ - hast du ein Bild oder eine Zeichnung wo die Abschirmung vom SMA Kabel angelötet werden muss? Ich würde auch gerne die externe Antenne verwenden.

**Comment by dayt47** [comment]
@jens-maus Ich hatte mal gelesen, dass der RPI-RF-MOD die Signale entstört und stabiler läuft als der alte HM-MOD-RPI-PCB. Wie sieht es aber mit dem HmIP-RFUSB aus? Ist letzterer besser als der RPI-RF-MOD oder sind die gleichwertig mit externer Antenne?

**Reply by jens-maus** [reply]
Das Ganze (d.h. das anlöten einer externen Antenne oder SMA Pigtail) läuft bei den Homematic Funkmodulen/Coprozessoren immer nach dem gleichen Schema ab. Siehe z.B. hier:

https://homematic-forum.de/forum/viewtopic.php?f=27&t=27287

D.h. den Signaldraht da anlöten wo aktuell der Antennenstummel des Funkmodules (bei dir ein HM-MOD-RPI-PCB) angelötet ist und die Schirmung (GND) am besten direkt an das Schirmblech des Funkmodules anlöten.

Bzgl. HmIP-RFUSB vs RPI-RF-MOD besser für externe Antenne. Würde ich vermuten gleichwertig, käme aber auf einen Versuch an.

**Comment by Baxxy13** [comment]
Hier mal mein HmIP-RFUSB im HAP-Gehäuse.
Die Schirmung des Pigtails ist an der Kante des Funkmoduls (halbrundes Lötauge) angelötet und das ganze mit ordentlich Heißkleber fixiert.
![RFUSB in HAP Gehäuse](https://github.com/jens-maus/RaspberryMatic/assets/54898249/ba388cac-0e61-4041-af1f-75a6746afffe)

**Reply by AlexanderDedy** [reply]
Vielen Dank! Ich bestelle jetzt den Stick und das Kabel. Ich kann es leider auf dem Bild nicht gut erkennen.. ist der Lötpunkt für die Abschirmung auf der kleinen oder größeren Platine?

**Reply by Baxxy13** [reply]
Auf der kleinen, dem eigentlichen Funkmodul. Ja, ist verdeckt. Aber man sieht die gleiche "Kerbe" ja nochmal links oberhalb der angelöteten Seele.
